File: braggbnb108_ui/src/services/CancellationPolicyService.js
import http from "../http-common"; 

class CancellationPolicyService {
  getAllCancellationPolicys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/cancellationPolicy/cancellationPolicys`, searchDTO);
  }

  get(cancellationPolicyId) {
    return this.getRequest(`/cancellationPolicy/${cancellationPolicyId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/cancellationPolicy?field=${matchData}`, null);
  }

  addCancellationPolicy(data) {
    return http.post("/cancellationPolicy/addCancellationPolicy", data);
  }

  update(data) {
  	return http.post("/cancellationPolicy/updateCancellationPolicy", data);
  }
  
  uploadImage(data,cancellationPolicyId) {
  	return http.postForm("/cancellationPolicy/uploadImage/"+cancellationPolicyId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CancellationPolicyService();

File: src/main/java/com/braggbnb108/service/CancellationPolicyService.java
package com.braggbnb108.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb108.domain.CancellationPolicy;
import com.braggbnb108.dto.CancellationPolicyDTO;
import com.braggbnb108.dto.CancellationPolicySearchDTO;
import com.braggbnb108.dto.CancellationPolicyPageDTO;
import com.braggbnb108.dto.CancellationPolicyConvertCriteriaDTO;
import com.braggbnb108.service.GenericService;
import com.braggbnb108.dto.common.RequestDTO;
import com.braggbnb108.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CancellationPolicyService extends GenericService<CancellationPolicy, Integer> {

	List<CancellationPolicy> findAll();

	ResultDTO addCancellationPolicy(CancellationPolicyDTO cancellationPolicyDTO, RequestDTO requestDTO);

	ResultDTO updateCancellationPolicy(CancellationPolicyDTO cancellationPolicyDTO, RequestDTO requestDTO);

    Page<CancellationPolicy> getAllCancellationPolicys(Pageable pageable);

    Page<CancellationPolicy> getAllCancellationPolicys(Specification<CancellationPolicy> spec, Pageable pageable);

	ResponseEntity<CancellationPolicyPageDTO> getCancellationPolicys(CancellationPolicySearchDTO cancellationPolicySearchDTO);
	
	List<CancellationPolicyDTO> convertCancellationPolicysToCancellationPolicyDTOs(List<CancellationPolicy> cancellationPolicys, CancellationPolicyConvertCriteriaDTO convertCriteria);

	CancellationPolicyDTO getCancellationPolicyDTOById(Integer cancellationPolicyId);







}





